flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/RowSerializerUpgradeTest.java

```diff
@@ -76,14 +84,16 @@ public RowSerializerUpgradeTest(TestSpecification<Row, Row> testSpecification) {
 	public static final class RowSerializerSetup implements TypeSerializerUpgradeTestBase.PreUpgradeSetup<Row> {
 		@Override
 		public TypeSerializer<Row> createPriorSerializer() {
-			return stringLongRowSupplier();
+			return createRowSerializer(true);
```

Comment:
I think there's a mis-understanding in the use of this `createPriorSerializer` method.
This method is only ever used when writing the test snapshot files, under the target branch.

e.g.
- if we checkout to the `release-1.10` branch and generate a snapshot, this should create a serializer that writes in the old format. 
- if we checkout to the `release-1.11` branch and generate a snapshot, this should create a serializer that writes in the new format.

With the current changes in the PR, the generated snapshot files will always be written with the old format, regardless of which branch you're on.

```suggestion
             // in older branches, this writes in old format;
             // in newer branches >= 1.11, this writes in new format
			return new RowSerializer(fieldSerializers);
```

Reply:
To really clarify this, I think we should make the `RowSerializer` constructor that allows passing in the `legacyModeEnabled` flag private, to be only usable by the `RowSerializerSnapshot#createOuterSerializer`. This concern should not be leaked into tests.

The bottom line is, the concern of creating an old serializer with previous formats should only be visible to the snapshots.